### ISSUE TITLE
Refactoring into webextension

### ISSUE THREAD

**Issue by kirlat** [alpheios-project/safari-app-front#4]
*Based on discussion in Slack:*

Since `safari-app-front` and `webextension` shares a majority of the code, it would be good to merge them into one directory. This will simplify further development and code maintenance. Also, since `safari-app-back` serves the same function as webextinsion's background script, it would make sense to place them side by side too, even though their code base would remain separated.

To simplify the process and minimize risks we can do this as a two-phase approach:

**PHASE ONE**
* Move all the code from `safari-app-front` into a subdirectory called `content-safari` within a `webextension/src` dir. Update links within `content-safari` as necessary.
* Create new Safari build targets (`build-prod-safari` and `build-dev-safari`) that will build code from `content-safari` and put a resulting bundle into a `webextension/dist` directory. This bundle should have a `-safari` postfix to distinguish it from the webextension's content builds.
* Move all the code from `safari-app-back` into a subdirectory called `background-safari` within a `webextension/src` dir. Update XCode project links to point to CSS and JS files from within a `webextension/dist` directory (those file should be included into the project as the ones with relative path).

**PHASE TWO**
(can be done later during refactoring)
Merge the code base from `content-safari` with the code from `content` as much as possible.

We can discuss further implementation details as needed.

**Comment by balmas** [comment]
yes I agree with proceeding with this approach.  @irina060981 can you take a stab at making this change? And @kirlat please provide further guidance as needed. Thanks both!

**Comment by irina060981** [comment]
First step 
move safari-app-front to webextension repo and update build

1) update components to handle with webextension and safari app front - not to render iframe for safari
https://github.com/alpheios-project/components/pull/260

2) update webextension 
https://github.com/alpheios-project/webextension/pull/120

3) update safari-app-back to handle with new names of the content part
without PR - https://github.com/alpheios-project/safari-app-back

**Comment by kirlat** [comment]
@irina060981, great updates! Can you also please move `safari-app-back` content into `webextension/src/background-safari`? Thanks!

**Comment by irina060981** [comment]
First step is merged.
@kirlat, yes - it is planned as the second step :)

**Comment by irina060981** [comment]
This PR alpheios-project/components#260 - is cancelled upon @kirlat suggestion.

**Comment by kirlat** [comment]
> @kirlat, yes - it is planned as the second step :)

Great, thanks!

**Comment by irina060981** [comment]
I have uploaded background part to webextension.
I have signed the application as a member of Alpheios Team and pack with signiture to dist/SafariBuild/SafariBuild.dmg

**Comment by irina060981** [comment]
I have made the third step - refactor to reduce duplicate code
Also I have renamed it more correctly, changed reference from me to Alpheios in project and targets properties (where I was able to find, may be there are some other places - if I find I will correct too).
Rebuild and rearchived, changed the description and version that is shown in Safari Extensions Tab.

https://github.com/alpheios-project/webextension/pull/123

P.S. @kirlat and @balmas , there are some problems with GitHub today - so I am not sure that I have added you both as reviewers - because I couldn't open the PR and coudn't create a new one.
I hope these problems will be solved later bu GitHub team.
